app/helpers/course/users_helper.rb

```diff
@@ -1,4 +1,24 @@
 # frozen_string_literal: true
 module Course::UsersHelper
-  include Course::Achievement::AchievementsHelper
+  # Returns a hash that maps +User+ ids to their +CourseUser+ IDs in a given +course_id+
+  #
+  # @param [Integer] course_id The ID of the course
+  # @return [Hash]
+  def preload_course_user_ids_hash(course_id)
+    course_users_in(course_id).to_h { |course_user| [course_user.user_id, course_user.id] }
+  end
+
+  # Returns a hash that maps +User+ ids to their +CourseUser+ in a given +course_id+
+  #
+  # @param [Integer] course_id The ID of the course
+  # @return [Hash]
+  def preload_course_users_hash(course_id)
```

Comment:
I think you can pass in `course` instead of `course_id` and then you can load the course users by association (eg `course.course_users`